CRM/Admin/Form/MailSettings.php

```diff
@@ -108,11 +126,15 @@ public function getDefaultEntity() {
   public function setDefaultValues() {
     $defaults = parent::setDefaultValues();
 
-    // Set activity status to "Completed" by default.
-    if ($this->_action != CRM_Core_Action::DELETE &&
-      (!$this->_id || !CRM_Core_DAO::getFieldValue('CRM_Core_BAO_MailSettings', $this->_id, 'activity_status'))
-    ) {
+    if (!$this->_id) {
+      $defaults['is_ssl'] = TRUE;
+      $defaults['is_default'] = 0;
+      $defaults['activity_type_id'] = CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_Activity', 'activity_type_id', 'Inbound Email');
       $defaults['activity_status'] = 'Completed';
+      $defaults['activity_source'] = 'from';
+      $defaults['activity_targets'] = 'to,cc,bcc';
+      $defaults['activity_assignees'] = 'from';
+      $defaults['is_active'] = TRUE;
     }
```

Comment:
Looking at how the parent form works, how about (plus the upgrade change for bounce acct)

```suggestion
    $defaults['is_ssl'] = $defaults['is_ssl'] ?? TRUE;
    $defaults['is_default'] = $defaults['is_default'] ?? 0;
    $defaults['activity_type_id'] = $defaults['activity_type_id'] ?? CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_Activity', 'activity_type_id', 'Inbound Email');
    $defaults['activity_status'] = $defaults['activity_status'] ?? 'Completed';
    $defaults['activity_source'] = $defaults['activity_source'] ?? 'from';
    $defaults['activity_targets'] = $defaults['activity_targets'] ?? 'to,cc,bcc';
    $defaults['activity_assignees'] = $defaults['activity_assignees'] ?? 'from';
    $defaults['is_active'] = $defaults['is_active'] ?? TRUE;
```

Seems to work for me.

Reply:
Yes, this is perfect, have done this and updated the upgrade so that it sets the default values for bounce processing as well.